docs/deploying_an_application.md

```diff
@@ -0,0 +1,109 @@
+# Empire :: Deploying an Application
+
+Before using Empire successfully it's important to understand it's design and architecture. It's suggested you first read the `Features & Design Goals document`.
+
+There are a few key things which your applications, and overall architecture, will need to keep in mind in order to be successful with Empire.  Below is an exhaustive set of criteria your applications must meet in order to work with Empire.  It's not that long and by no means locks you in to Empire.  Structing your application in this way provides many benefits, only one of which is being able to be managed by Empire.
+
+For the impatient, there are two absolutes:
+
+- Your must create a `Procfile` which defines how to run your application
+- If you application answers requests over the network, it must listen on port 8080, or better, on the `$PORT` environment variable.
+
+To see an example application, you can look at [remind101/acme-inc]
+
+
+
+## Procfile
+
+Empire was modeled after the Heroku API.  As such, many concepts and commands are similar.  One concept which has bled over and which is a requirement is that of a [Procfile][procfiles].  For your application to actually work, you must include a Procfile at the root level of your appplication.
+
+Imagine a Python application which simply ran celery.  A `Procfile` for such a service could look like this:
+
+```
+workerservice: celery -A tasks worker --loglevel=info 
+```
+
+The file would be named `Procfile`, and live at the directory root for your application.
+
+```console
+$ tree .
+.
+├── Dockerfile
+├── Procfile
+├── myapplication/ 
+
+```
+
+
+## Empire applicaiton types
+
+Empire treats "web" and "non-web" services different.  Here, a "web" process is defined as anyting which needs to expose a port.  The way to differentiate the two types of processes is easy.  
+
+### Non-web processes
+
+Let's start with non-web processes first since they're much simpler.  If your service doens't expose a port it doesn't need to be discovered by any other system.  With that, all you need to do it deploy your application and let Empire manage it. The act of discovering *other* services is up to you (database, caches, etc.)
+
+### Web processes
+
+If your service is going to be used by other systems, you'll need to run some type of server which exposes a port.  In this scenario you'll need to name it `web` in your Procfile:
+
+```
+web: node server.js
+```
+
+or, using Django as an example:
+
+```
+web: python manage.py runserver
+```
+
+For web applications, Empire does a few things.  There are multiple layers of routing, none of which are extrelemy complex.
+
+Assume we have deployed a container named `mycompany/awesome-app`.
+
+The routing to your application is handled as such inside of the VPC:
+
+```
+http://awesome-app --> ELB to minions (port 80 to [9000-10000]) --> Host port [9000-10000] -> container port 8080
```

Comment:
I feel like this could be a little more clear.  Maybe:

```
http://awesome-app -> ELB (port 80) -> Minion (port [9000-10000]) -> container port $PORT (8080)
```

??